From 1276f09bae331988420e9e0a61184a631b8734d1 Mon Sep 17 00:00:00 2001
From: De Huo <De.Huo@windriver.com>
Date: Sat, 3 Aug 2019 18:27:44 +0800
Subject: [PATCH] Add WRLinux specific rule package_strongswan_installed

Add WRLinux specific rule package_strongswan_installed, since WRLinux1019 use 
this package to regulate ipsec.

Upstream-Status: Pending

Signed-off-by: De Huo <De.Huo@windriver.com>
---
 .../oval/wrlinux1019.xml                           | 104 +++++++++++++++++++++
 .../package_strongswan_installed/rule.yml          |  18 ++++
 .../profiles/draft_stig_wrlinux_disa.profile       |   1 +
 3 files changed, 123 insertions(+)
 create mode 100644 linux_os/guide/system/network/network-ipsec/package_strongswan_installed/oval/wrlinux1019.xml
 create mode 100644 linux_os/guide/system/network/network-ipsec/package_strongswan_installed/rule.yml

diff --git a/linux_os/guide/system/network/network-ipsec/package_strongswan_installed/oval/wrlinux1019.xml b/linux_os/guide/system/network/network-ipsec/package_strongswan_installed/oval/wrlinux1019.xml
new file mode 100644
index 0000000..745ca71
--- /dev/null
+++ b/linux_os/guide/system/network/network-ipsec/package_strongswan_installed/oval/wrlinux1019.xml
@@ -0,0 +1,104 @@
+<def-group>
+  <definition class="compliance" id="package_strongswan_installed"
+  version="1">
+    <metadata>
+      <title>Package strongswan Installed</title>
+      <affected family="unix">
+        <platform>multi_platform_wrlinux</platform>
+      </affected>
+      <description>The strongswan package is installed</description>
+    </metadata>
+    <criteria operator="AND">
+      <criterion comment="The /etc/ipsec.conf file is present" test_ref="test_etc_ipsec_conf_is_present" />
+      <criterion comment="The /etc/strongswan.conf file is present" test_ref="test_etc_strongswan_conf_is_present" />
+      <criterion comment="The /lib/systemd/system/strongswan-starter.service file is present" test_ref="test_lib_systemd_system_strongswan_service_is_present" />
+      <criterion comment="The /usr/bin/pki file is present" test_ref="test_usr_bin_pki_is_present" />
+      <criterion comment="The /usr/sbin/ipsec file is present" test_ref="test_usr_sbin_ipsec_is_present" />
+    </criteria>
+  </definition>
+
+  <unix:file_test id="test_etc_ipsec_conf_is_present"
+	check="all"
+	check_existence="all_exist"
+	comment="The /etc/ipsec.conf file is present on the system"
+	version="1"
+	>
+    <unix:object object_ref="object_etc_ipsec_conf_is_present" />
+  </unix:file_test>
+
+  <unix:file_object id="object_etc_ipsec_conf_is_present"
+	comment="The /etc/ipsec.conf file is present on the system"
+	version="1"
+	>
+    <unix:filepath operation="equals">/etc/ipsec.conf</unix:filepath>
+  </unix:file_object>
+
+
+  <unix:file_test id="test_etc_strongswan_conf_is_present"
+	check="all"
+	check_existence="all_exist"
+	comment="The /etc/strongswan.conf file is present on the system"
+	version="1"
+	>
+    <unix:object object_ref="object_etc_strongswan_conf_is_present" />
+  </unix:file_test>
+
+  <unix:file_object id="object_etc_strongswan_conf_is_present"
+	comment="The /etc/strongswan.conf file is present on the system"
+	version="1"
+	>
+    <unix:filepath operation="equals">/etc/strongswan.conf</unix:filepath>
+  </unix:file_object>
+
+
+  <unix:file_test id="test_lib_systemd_system_strongswan_service_is_present"
+	check="all"
+	check_existence="all_exist"
+	comment="The /lib/systemd/system/strongswan-starter.service file is present on the system"
+	version="1"
+	>
+    <unix:object object_ref="object_lib_systemd_system_strongswan_service_is_present" />
+  </unix:file_test>
+
+  <unix:file_object id="object_lib_systemd_system_strongswan_service_is_present"
+	comment="The /lib/systemd/system/strongswan-starter.service file is present on the system"
+	version="1"
+	>
+    <unix:filepath operation="equals">/lib/systemd/system/strongswan-starter.service</unix:filepath>
+  </unix:file_object>
+
+
+  <unix:file_test id="test_usr_bin_pki_is_present"
+	check="all"
+	check_existence="all_exist"
+	comment="The /usr/bin/pki file is present on the system"
+	version="1"
+	>
+    <unix:object object_ref="object_usr_bin_pki_is_present" />
+  </unix:file_test>
+
+  <unix:file_object id="object_usr_bin_pki_is_present"
+	comment="The /usr/bin/pki file is present on the system"
+	version="1"
+	>
+    <unix:filepath operation="equals">/usr/bin/pki</unix:filepath>
+  </unix:file_object>
+
+
+  <unix:file_test id="test_usr_sbin_ipsec_is_present"
+	check="all"
+	check_existence="all_exist"
+	comment="The /usr/sbin/ipsec file is present on the system"
+	version="1"
+	>
+    <unix:object object_ref="object_usr_sbin_ipsec_is_present" />
+  </unix:file_test>
+
+  <unix:file_object id="object_usr_sbin_ipsec_is_present"
+	comment="The /usr/sbin/ipsec file is present on the system"
+	version="1"
+	>
+    <unix:filepath operation="equals">/usr/sbin/ipsec</unix:filepath>
+  </unix:file_object>
+
+</def-group>
diff --git a/linux_os/guide/system/network/network-ipsec/package_strongswan_installed/rule.yml b/linux_os/guide/system/network/network-ipsec/package_strongswan_installed/rule.yml
new file mode 100644
index 0000000..34d30f1
--- /dev/null
+++ b/linux_os/guide/system/network/network-ipsec/package_strongswan_installed/rule.yml
@@ -0,0 +1,18 @@
+documentation_complete: true
+
+prodtype: wrlinux1019
+
+title: 'Install the strongswan Package'
+
+description: |-
+    The <tt>strongswan</tt> package should be installed.
+    {{{ describe_package_install(package="strongswan") }}}
+
+rationale: |-
+    The strongswan package must be installed if it is to be available for integrity checking.
+
+severity: medium
+
+ocil_clause: 'the package strongswan is not installed'
+
+ocil: '{{{ ocil_package(package="strongswan") }}}'
diff --git a/wrlinux1019/profiles/draft_stig_wrlinux_disa.profile b/wrlinux1019/profiles/draft_stig_wrlinux_disa.profile
index 6e5a73d..c2a6f3f 100644
--- a/wrlinux1019/profiles/draft_stig_wrlinux_disa.profile
+++ b/wrlinux1019/profiles/draft_stig_wrlinux_disa.profile
@@ -235,3 +235,4 @@ selections:
     - samhain_periodic_cron_checking
     - samhain_scan_notification
     - package_openssh-server_installed
+    - package_strongswan_installed
-- 
1.9.1

